lib/puppet/provider/mysql.rb

```diff
@@ -38,11 +38,35 @@ class Puppet::Provider::Mysql < Puppet::Provider
   ].join(':')
```

Comment:
I'm looking at this horrible hack above and wonder if it should use the proper API. Something like this probably works:
```ruby
has_command(:mysql_client, 'mysql') do
  environment PATH: path, LD_LIBRARY_PATH: ld_library_path
end
```

You can also use `is_optional` according to https://github.com/puppetlabs/puppet/blob/e227c27540975c25aa22d533a52424a9d2fc886a/lib/puppet/provider.rb#L177-L214.